connect/runtime/src/main/java/org/apache/kafka/connect/runtime/WorkerSinkTask.java

```diff
@@ -139,25 +139,30 @@ public void execute() {
     }
 
     protected void iteration() {
-        long now = time.milliseconds();
-
-        // Maybe commit
-        if (!committing && now >= nextCommit) {
-            commitOffsets(now, false);
-            nextCommit += workerConfig.getLong(WorkerConfig.OFFSET_COMMIT_INTERVAL_MS_CONFIG);
-        }
+        long timeoutMs;
+        if (!context.offsetCommitDisabled()) {
+            long now = time.milliseconds();
+
+            // Maybe commit
```

Comment:
Would it make sense to pull this block into a separate function (e.g. `maybeCommitOffsets()`)?